Billions - miliardy

ipepe opened the pull request:
I added billions support

ipepe commented:
I don't know why travis is reporting errors? Could anyone tell me what should I fix?

wpiekutowski commented:
Thanks for the PR. I'll take a look at the travis issue and code itself, but first please fix the following cosmetic issues:
- use only English
- close the previous PR
- squash all code climate commits together
- squash changelog commits together
- remove the merge commit

ipepe commented:
Well in my commits I added only billions support. That is why I added it as separate PR

ipepe commented:
@wpiekutowski 
I did all You asked about, except closing PR because I'm not owner of it.

Commit:
Added billions support.
Fixed multiple bugs.
Added changelog to readme.
Added support for genders in currencies.
Added support for hundredth parts of number.(setne i grosze)
Remove "jeden" from million and thousand when only one.
Added support for adding new currencies.

wpiekutowski commented:
Thank you very much. I probably won't be able to take care of this until
Monday though, so please be patient.

On 31 March 2016 at 15:41, Patryk Ptasiński notifications@github.com
wrote:

> @wpiekutowski https://github.com/wpiekutowski
> I did all You asked about, except closing PR because I'm not owner of it.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> https://github.com/amberbit/polish_number/pull/5#issuecomment-203942727

ipepe commented:
@wpiekutowski How it is going? Becasue we would like to update Gem without pointing to our private repositories

wpiekutowski commented on .gitignore:
```diff
@@ -1,3 +1,4 @@
 *.gem
 Gemfile.lock
 .bundle
+.idea
```
Missing new line.

wpiekutowski commented on .ruby-version:
```diff
@@ -1 +1 @@
-2.1.5
+2.1.5
```
Missing new line.

wpiekutowski commented on README.md:
```diff
@@ -26,6 +26,35 @@ Or install it yourself as:
 
+## Cents format
+
+    PolishNumber.translate(34.11) #=> trzydzieści cztery i jedenaście setnych
```
Do you actually use floats to store currency amounts?

wpiekutowski commented:
@ipepe you've squashed all commits together. This means that @simcha authorship information is lost. @simcha is that ok with you?

wpiekutowski commented on README.md:
```diff
@@ -26,6 +26,35 @@ Or install it yourself as:
+
+    PolishNumber.add_currency(:COWS, { :one => 'krowa', :few => 'krowy',
+      :many => 'krów', :gender => :she,
+      :one_100 => 'ser', :few_100 => 'sery',
```
`:one_cent`, `:few_cents`, `:many_cents` seem to be easier to understand.

wpiekutowski commented:
I'd like to work with both of you to merge this PR. I think it could be useful for many people working on projects with Polish language.

wpiekutowski commented on README.md:
```diff
@@ -26,6 +26,35 @@ Or install it yourself as:
+    PolishNumber.translate(0.10, :currency => :PLN) #=> dziesięć groszy
+    PolishNumber.translate(5.10, :currency => :PLN, :cents => :no ) #=> pięć złotych
+    PolishNumber.translate(5, :currency => :PLN, :cents => :words ) #=> pięć złotych zero groszy
+    PolishNumber.translate(5, :currency => :PLN, :cents => :digits ) #=> pięć złotych 00/100
```
`:cents` option is a good idea. Can we rename this value to something that more closely represents true meaning? For example `:fraction`?

wpiekutowski replied:
Unneeded space before `)`

wpiekutowski commented on README.md:
```diff
@@ -26,6 +26,35 @@ Or install it yourself as:
+    PolishNumber.translate(34.11) #=> trzydzieści cztery i jedenaście setnych
+    PolishNumber.translate(30.01, :currency => :PLN) #=> trzydzieści złote i jeden grosz
+    PolishNumber.translate(0.10, :currency => :PLN) #=> dziesięć groszy
+    PolishNumber.translate(5.10, :currency => :PLN, :cents => :no ) #=> pięć złotych
```
Can we rename this value to `:none` please?

wpiekutowski replied:
Unneeded space before `)`

wpiekutowski commented on README.md:
```diff
@@ -26,6 +26,35 @@ Or install it yourself as:
+## Adding new currencies
+
+    PolishNumber.add_currency(:COWS, { :one => 'krowa', :few => 'krowy',
+      :many => 'krów', :gender => :she,
```
Can we use masculine, feminine and neuter for gender value?

wpiekutowski commented on dev_console.sh:
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+
+# Run the console with all the code loaded
+pry -I "./lib" -r "./lib/polish_number"
```
Missing new line.

wpiekutowski commented on README.md:
```diff
@@ -26,6 +26,35 @@ Or install it yourself as:
+    PolishNumber.translate(30.01, :currency => :PLN) #=> trzydzieści złote i jeden grosz
+    PolishNumber.translate(0.10, :currency => :PLN) #=> dziesięć groszy
+    PolishNumber.translate(5.10, :currency => :PLN, :cents => :no ) #=> pięć złotych
+    PolishNumber.translate(5, :currency => :PLN, :cents => :words ) #=> pięć złotych zero groszy
```
Unneeded space before `)`

wpiekutowski commented on README.md:
```diff
@@ -26,6 +26,35 @@ Or install it yourself as:
+
+## Adding new currencies
+
+    PolishNumber.add_currency(:COWS, { :one => 'krowa', :few => 'krowy',
```
Unneeded space after `{`. Please improve formatting of this part.

wpiekutowski replied:
This isn't a currency. It's a unit. Can we keep `:currency` in `.translate` for backwards compatibility? Can we add there `:unit` as an alias to `:currency`? Can we rename this method to `.add_unit`?

wpiekutowski commented on lib/polish_number.rb:
```diff
@@ -21,86 +21,181 @@ module PolishNumber
+
+  CENTS = [:auto, :no, :words, :digits]
+
   CURRENCIES = {
```
Again `UNITS` would make much more sense, if you want to have support for arbitrary words, like in the next line.

wpiekutowski commented on lib/polish_number.rb:
```diff
@@ -21,86 +21,181 @@ module PolishNumber
+
   CURRENCIES = {
-    :PLN => {:one => 'złoty', :few => 'złote', :many => 'złotych'}
+    :NO => {:one => '', :few => '', :many => '',
```
Is NO a shortcut from NUMBER?

wpiekutowski commented on lib/polish_number.rb:
```diff
@@ -21,86 +21,181 @@ module PolishNumber
+  end
+
+  def self.add_currency(name, hash)
+    CURRENCIES[name]=hash
```
This isn't thread-safe. Perhaps we should use https://github.com/ruby-concurrency/thread_safe or immutable data structure?

wpiekutowski commented:
I see some other minor formatting issues, but we can tackle that after we figure out some more important things:
- using floats for currency amounts
- thread safety
- naming things

@ipepe I'd like to push here a fix to travis CI issue. Could you give me access to your fork?

Commit:
Fixed names and formatting

ipepe commented:
I fixed some formatting and naming issues that You addressed. But I can't help You more in this matter. Also I gave You permission to my repository

wpiekutowski commented:
Thanks!

wpiekutowski commented:
I'm closing this because multiple issues with this PR were never fixed.
